From 8798741ed602ca2459ab22982d4ecdad00ec4c2b Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 11 Sep 2018 14:35:44 +0000
Subject: [PATCH 0845/1000] MANTIS 0041370 Enhancement of Resolution Settings

appcontrol.idl
V1.5

- added documentation for command and indication
- added resolutionEx with width and height
- added CMD_SET_RESOLUTION_EX to set resolution with resolutionEx
- added function supportedResolutions to return the list of supported resolutionEx


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1375 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/appcontrol.idl          | 61 +++++++++++++++----
 1 file changed, 48 insertions(+), 13 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl b/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl
index c44fdfc2..47292919 100644
--- a/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl
@@ -3,6 +3,13 @@
 // V1.0.0.0
 // application is deprecated; retained for compatibility with old MCC
 // applicationStart.cmdLine changed from String to List of Strings (more than one process per application allowed)
+// V1.5
+// SteT extension Mantis 41370: Enhancement of Resolution Settings
+// - added documentation for command and indication
+// - added resolutionEx with width and height 
+// - added CMD_SET_RESOLUTION_EX to set resolution with resolutionEx
+// - added function supportedResolutions to return the list of supported resolutionEx
+
 
 #ifndef APPCONTROL_IDL
 #define APPCONTROL_IDL
@@ -22,8 +29,26 @@
 
 module appControl
 {
-    const string interface_version = "1.4";
-   
+    const string interface_version = "1.5";
+
+/*
+using DeviceComponents::indicationListener and DeviceComponents::Generic from mdci.idl
+  RC indicationListener::indication(in any ind);
+    with ind: DeviceComponents::indication
+      with ind.type:   appControl::indicationType
+      with ind.status: DeviceComponents::nullData   for IND_RESOURCE_USAGE
+      with ind.data:   appControl::data             for IND_RESOURCE_USAGE
+  RC Generic::exec(in any cmd);
+    with cmd: DeviceComponents::command
+      with cmd.type: appControl::commandType
+      with cmd.data: appControl::application        for CMD_START
+      with cmd.data: appControl::applicationId      for CMD_SWITCH, CMD_KILL, CMD_SHOW
+      with cmd.data: appControl::resolutions        for CMD_SET_RESOLUTION
+      with cmd.data: appControl::displayState       for CMD_SWITCH_DISPLAY
+      with cmd.data: appControl::resolutionEx       for CMD_SET_RESOLUTION_EX
+      with cmd.data: DeviceComponents::nullData     for CMD_KILLALL, CMD_SWITCHMAIN
+*/
+
     // list of strings
     typedef sequence<string> strings;
 
@@ -38,18 +63,26 @@ module appControl
       RES_1600x1200,
       RES_DEFAULT
     };
-    
+
     enum displayState
     {
       DISPLAY_ON,
       DISPLAY_OFF
     };
 
+    struct resolutionEx
+    {
+      long width;
+      long height;
+    };
+    
+    typedef sequence<resolutionEx> resolutionExList;
+
     // deprecated for CMD_START used applicationStart, for CMD_SWITCH, CMD_KILL, CMD_SHOW use applicationId */
     struct application
     {
       string appName;     /**< The application name as internally handeled by the ApplicationManager*/
-      string cmdLine;     /**< ignored; */ 
+      string cmdLine;     /**< ignored; */
     };
 
     /** Defines the application data that is passed by the ApplicationManager to this device
@@ -75,14 +108,15 @@ module appControl
 
     enum commandType
     {
-      CMD_START,          /**< Starts a new application in a new desktop environment */
-      CMD_SWITCH,         /**< Switches to a specific desktop */
-      CMD_KILL,           /**< terminates the application and closes the associated desktop */
-      CMD_KILLALL,        /**< terminates all applications and closes the associated desktops */
-      CMD_SWITCHMAIN,     /**< Switches to the main desktop */
-      CMD_SHOW,           /**< Switches to a specific desktop, but doesn't activate it (debug mode)*/
-      CMD_SET_RESOLUTION, /**< Set a screen resolution */
-      CMD_SWITCH_DISPLAY  /**< Switch display ON or OFF */
+      CMD_START,                    /**< Starts a new application in a new desktop environment */
+      CMD_SWITCH,                   /**< Switches to a specific desktop */
+      CMD_KILL,                     /**< terminates the application and closes the associated desktop */
+      CMD_KILLALL,                  /**< terminates all applications and closes the associated desktops */
+      CMD_SWITCHMAIN,               /**< Switches to the main desktop */
+      CMD_SHOW,                     /**< Switches to a specific desktop, but doesn't activate it (debug mode)*/
+      CMD_SET_RESOLUTION,           /**< Set a screen resolution */
+      CMD_SWITCH_DISPLAY,           /**< Switch display ON or OFF */
+      CMD_SET_RESOLUTION_EX         /**< Set a screen resolution */
     };
 
     /** All device components inherit from the generic DeviceComponents interface */
@@ -92,6 +126,7 @@ module appControl
       /** returns the current screen resolution */
 
       resolutions currentResolution();
+      resolutionExList supportedResolutions();
     };
 
     enum indicationType
@@ -99,7 +134,7 @@ module appControl
       IND_RESOURCE_USAGE,
       IND_UNDEFINED
     };
-    
+
     struct data
     {
       string applicationName;
-- 
2.41.0.windows.1

